Update visualization design pattern (#273)
pxblue/doc-it: src/docs/patterns/visualizations.mdx

import { DemoCard, ImageGrid, MaterialDesignDescription, Divider } from '../../app/components';

import VisualizationBanner from '../../app/assets/design-patterns/visualizations/visualizations-banner.svg';
import LineChart from '../../app/assets/design-patterns/visualizations/line-chart.png';
import PieChart from '../../app/assets/design-patterns/visualizations/pie-chart.png';
import ClockwisePie from '../../app/assets/design-patterns/visualizations/cw-ccw.png';
import MapsImg from '../../app/assets/design-patterns/visualizations/map.png';
import DontUsePrimaryColor from '../../app/assets/design-patterns/visualizations/use-primary-color.png';

<!-- KEYWORDS: design pattern visualization map chart highcharts chartjs mapbox -->

# Visualizations

> <ImageGrid regularWidth images={[VisualizationBanner]} />

Reviewer: I like the right side of this, but the axis pieces make it feel like a lot of empty space.

This also violates our principle below about using the branding colors for charts 😆 , but I think it's okay to match the style of the other banners.

Reviewer: The other banners feel like they have a little more 'depth' whereas this one feels very flat...can we do something with a background or a legend?

> There are many opportunities to visualize data in different applications. The most common visualizations used in PX Blue applications come in the form of charts / graphs and maps.

Reviewer: ```suggestion
Data is the foundation of most applications, but many users are not interested in looking at the raw data. Visualizations allow you to present this data in a (visual) way that is easier to understand and allows users to draw conclusions and take actions based on the data. The most common visualizations in PX Blue applications are charts / graphs and maps.
```

<Divider />

# Charts / Graphs

<ImageGrid
    images={[LineChart, PieChart]}

>     caption={'Line charts (#1) and pie / ring charts (#2) are the most frequently used data visualizations in PX Blue.'}

Reviewer: ```suggestion
    caption={['1. Line charts represent time-series data changing over a period of time.', '2. Pie / ring charts show a single value as a proportion of an expected or ideal total.']}
```

/>

> There are [many variations](https://material.io/design/communication/data-visualization.html#types) of data visualizations. They can be a big line chart with 8 series, occupying the entire screen; or they can be as tiny as a simple pie chart embedded in [lists](/patterns/lists).

Reviewer: ```suggestion
Charts and graphs come in many different [forms](https://material.io/design/communication/data-visualization.html#types). Depending on the expected interaction, these can be large elements on the UI — such as full-screen line chart with numerous series — or small supporting elements like a pie chart embedded in a card.
```

<MaterialDesignDescription
    title={'Data Visualization'}
    url={'https://material.io/design/communication/data-visualization.html'}
/>

> > The following examples merely serve to show how to bring the charting libraries into your application and adjust some of the common chart parameters. You will want to adjust these values to meet the design requirements of your application.

Reviewer: Can we get rid of this and add a story to the backlog to revisit the design of these examples to follow our recommendations?

Author: PX Bleu 1613

<DemoCard
    repository={'highcharts'}
    angular={`https://stackblitz.com/github/pxblue/highcharts/tree/master/angular-demo`}
    react={`https://codesandbox.io/s/github/pxblue/highcharts/tree/master/react-demo`}
/>
<DemoCard
    repository={'chartjs'}
    angular={`https://stackblitz.com/github/pxblue/chartjs/tree/master/angular-demo`}
    react={`https://codesandbox.io/s/github/pxblue/chartjs/tree/master/react-demo`}
/>

> ## Common Pitfalls

Reviewer: There's only one sentence in this sections (and I don't know that I would necessarily call it a pitfall). I would just move that sentence up and have it under the umbrella of Charts/Graphs for now.

> <ImageGrid images={[ClockwisePie]} regularWidth caption={'A pie chart should always be clockwise.'} />

Reviewer: Is this true if we are in RTL? Should line charts get flipped for RTL? 

If so, is it worth linking to the internationalization page from this one somewhere?

Author: Mirror bar charts & line charts
No mirroring for anything circular (bc of the clock analogy)

> To avoid interfering with other UI elements, the charts should be built using primarily the [branding colors](/style/color#branding-colors).

Reviewer: ```suggestion
To avoid interfering with other UI elements, charts should be built using the PX Blue [branding](/style/color#branding-colors) color palette.
```

<React.Fragment>

>     <ImageGrid images={[DontUsePrimaryColor]} regularWidth caption={'Never use primary color palette in the charts.'} />

Reviewer: ```suggestion
    <ImageGrid images={[DontUsePrimaryColor]} regularWidth caption={'Never use the UI color palette for chart elements.'} />
```

    <Divider />
</React.Fragment>

# Maps

<ImageGrid
    regularWidth
    images={[MapsImg]}
    caption={[
        'An example of a map, built with PX Blue dark theme.',

>         'The embedded map is a screenshot from OpenStreetMap, licensed under CC-BY-SA 2.0.',

Reviewer: Do we need this caption at all?

Or maybe just "A simple geographic map using the dark theme."

Author: That credit line is necessary under the cc-by-sa license... 
The first line is updated to what you suggested.

    ]}
/>
<DemoCard
    repository={'mapbox'}
    angular={`https://stackblitz.com/github/pxblue/mapbox/tree/master/angular-demo`}
    react={`https://codesandbox.io/s/github/pxblue/mapbox/tree/master/react-demo`}
/>

> Maps can be used to indicate geographical locations. They typically span across an entire screen.

Reviewer: ```suggestion
Maps are used to represent spatial or geographical data. Typically they are used to plot sets of locations or facilities that a user is responsible for. 

Maps are usually full-screen elements and are interactive via scrolling, panning, zooming, etc.
```

> 

Reviewer: Is there anything we can add here or link out to to fill out this map section a little bit? It feels really light.

Author: Material Design does not have a proper page on the map topic.

<Divider />

# Developers

> There are many libraries and services for data visualizations. PX Blue currently provides themes for the following libraries:

Reviewer: ```suggestion
There are many libraries and services for data visualizations. PX Blue currently provides themes / utilities for the following libraries:
```
